File: GenerateInitvector/GenerateRandomKeyIvNonce.py
from Crypto.Random import get_random_bytes
import base64

def base64Encoding(input):
  dataBase64 = base64.b64encode(input)
  dataBase64P = dataBase64.decode("UTF-8")
  return dataBase64P

def generateRandomKey32Byte():
  return get_random_bytes(32)

def generateRandomIv():  
  return get_random_bytes(16)

def generateRandomNonce():  
  return get_random_bytes(12)

print("Generate a 32 byte long key, IV (16 byte), nonce (12 byte)")

aesKey = generateRandomKey32Byte()
aesKeyBase64 = base64Encoding(aesKey)
print("generated key length:    " + str(len(aesKey)) + " data: " + aesKeyBase64)

iv = generateRandomIv()
ivBase64 = base64Encoding(iv)
print("generated iv length:     " + str(len(iv)) + " data: " + ivBase64)

nonce = generateRandomNonce()
nonceBase64 = base64Encoding(nonce)
print("generated nonce length:  " + str(len(nonce)) + " data: " + nonceBase64)

File: GenerateInitvector/GenerateRandomKeyIvNonce.dart
import 'dart:convert';
import 'dart:math';
import 'dart:typed_data';
import "package:pointycastle/export.dart";

main() {
/* add in pubspec.yaml:
dependencies:
  pointycastle: ^3.1.1
 */

  printC('Generate a 32 byte long AES key');
  var aesKey = generateRandomAesKey();
  printC('aesKey (Base64): ' + base64Encoding(aesKey));

  printC('\nInitialization vector (IV)');
  var iv = generateRandomAesKey();
  printC('iv (Base64): ' + base64Encoding(iv));

  printC('\nGenerate a 12 byte long nonce for AES GCM');
  var nonce = generateRandomNonce();
  printC('nonce (Base64): ' + base64Encoding(nonce));
}

Uint8List generateRandomAesKey() {
  final _sGen = Random.secure();
  final _seed =
  Uint8List.fromList(List.generate(32, (n) => _sGen.nextInt(255)));
  SecureRandom sec = SecureRandom("Fortuna")..seed(KeyParameter(_seed));
  return sec.nextBytes(32);
}

Uint8List  generateRandomInitvector() {
  final _sGen = Random.secure();
  final _seed =
  Uint8List.fromList(List.generate(32, (n) => _sGen.nextInt(255)));
  SecureRandom sec = SecureRandom("Fortuna")..seed(KeyParameter(_seed));
  return sec.nextBytes(16);
}

Uint8List generateRandomNonce() {
  final _sGen = Random.secure();
  final _seed =
  Uint8List.fromList(List.generate(32, (n) => _sGen.nextInt(255)));
  SecureRandom sec = SecureRandom("Fortuna")..seed(KeyParameter(_seed));
  return sec.nextBytes(12);
}

String base64Encoding(Uint8List input) {
  return base64.encode(input);
}

void printC(String newString) {
  // for compatibility reasons to FlutterEmptyConsole
  print(newString);
}
